TESTS/netsocket/udp/udpsocket_echotest.cpp

```diff
@@ -109,6 +109,10 @@ void udpsocket_echotest_nonblock_receiver(void *receive_bytes)
             wait_ms(WAIT2RECV_TIMEOUT);
             --retry_cnt;
             continue;
+        } else if (recvd < 0) {
```

Comment:
What is actually achieved by this?

Reply:
in case an error status is returned, different than NSAPI_ERROR_WOULD_BLOCK we are raising a proper error, instead of just saying "data did not match"
This is how I understood this remark from Ari:
"UDPSOCKET_ECHOTEST_NONBLOCK/udpsocket_echotest_nonblock_receiver should also check for incorrect sizes."

Reply:
I would say that you should actually modify line 165. UDP datagram should always be send fully,  or report an error.

Reply:
Line 174: true.  Can't give any good reason why receiver reporting an error shouldn't be treated as test case failure so I'm willing to accept this PR.